src/Loop.php

```diff
@@ -430,3 +455,10 @@ private function __construct()
         // intentionally left blank
     }
 }
+
+// Reset the $level in order to be able to use ::run() inside a shutdown handler
+$f = function()
+{
+    self::$level = 0;
+};
+register_shutdown_function($f->bindTo(null, Loop::class));
```

Comment:
@bwoebi (1) I don't think we need `setFactory()` in shutdown hooks and introduce hacks for it. (2) I'd like to allow `setFactory` everywhere by removing the `$driver = null` reset here.